solr/core/src/java/org/apache/solr/cli/Tool.java

```diff
@@ -50,7 +50,19 @@ default String getFooter() {
     return "\nPlease see the Reference Guide for more tools documentation: https://solr.apache.org/guide/solr/latest/deployment-guide/solr-control-script-reference.html";
   }
 
-  List<Option> getOptions();
+  /**
```

Comment:
I find the getAllOptions and getOPtions very confusing, especially because we have logic in ToolBase that even goes through and uses getAllOptions and then iterates and checks that if they are deprecated and removes them.  I also think that the only use of the "getOptions" is in the help..   I'll try to refactor this and we can see if we like the commit or not!

Reply:
I was considering merging them too. You mentioning it now seems like the way to go. :)